### PR TITLE
Support for Hapi authentication

### PR DESCRIPTION
**Pull request by mirceaalexandru**

### CODE REVIEW

**Comment by mihaidma** [comment]
@mirceaalexandru I am reviewing this PR.

PR under review

File: .eslintrc [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
+  "extends": "standard",
```
Comment (geek) [inline .eslintrc]: Is there a reason to add standard?

File: auth.js [hunk 2]
```diff
@@ -3,732 +3,86 @@
...
 
-  seneca.add({role: 'auth', cmd: 'register_service'},
-    cmd_register_service)
-
-  seneca.add({role: 'auth', hook: 'map_fields'}, alias_fields)
-
-  function load_default_plugins () {
-    seneca.use(AuthToken)
-    seneca.use(AuthRedirect, options.redirect || {})
+    seneca.use(require('./lib/user-management'), internals.options)
```
Comment (geek) [inline auth.js]: Please add the require statements to the top of the file... it makes it easier to see what auth.js depends on

File: auth.js [hunk 3]
```diff
@@ -3,732 +3,86 @@
...
-      return respond(null, {ok: false, why: 'no-user'})
-    }
-
-    var user_data = msg.user
-    var q = {}
-    if (user_data.identifier) {
-      q[msg.service + '_id'] = user_data.identifier
-      user_data[msg.service + '_id'] = user_data.identifier
+  internals.choose_framework = function (){
+    if ('express' === internals.options.framework) {
```
Comment (geek) [inline auth.js]: Please move away from the yoda conditional style

```
if (\internals.options.framework === 'express') {
```

File: auth.js [hunk 4]
```diff
@@ -3,732 +3,86 @@
...
-  }
-
-  function _service_callback (service) {
-    return function (msg, respond) {
-      var req = msg.req$
-      var res = msg.res$
-      pp_auth[service](req, res, respond)
-    }
+  var m
+  if ((m = internals.options.prefix.match(/^(.*)\/+$/))) {
```
Comment (geek) [inline auth.js]: Please do the assignment first, then check if a match was found.

**Pull request closed without merging**